packages/camera/camera_avfoundation/ios/camera_avfoundation/Sources/camera_avfoundation/include/camera_avfoundation/CameraPlugin_Test.h

```diff
@@ -6,29 +6,41 @@
 
 #import "CameraPlugin.h"
 #import "FLTCam.h"
+#import "FLTCamConfiguration.h"
+#import "FLTCameraDeviceDiscovering.h"
+#import "FLTCaptureDevice.h"
 #import "messages.g.h"
 
+NS_ASSUME_NONNULL_BEGIN
+
+typedef id<FLTCaptureDevice> _Nonnull (^CaptureNamedDeviceFactory)(NSString *name);
+
 /// APIs exposed for unit testing.
 @interface CameraPlugin ()
 
 /// All FLTCam's state access and capture session related operations should be on run on this queue.
 @property(nonatomic, strong) dispatch_queue_t captureSessionQueue;
 
 /// An internal camera object that manages camera's state and performs camera operations.
-@property(nonatomic, strong) FLTCam *camera;
+@property(nonatomic, strong) FLTCam *_Nullable camera;
 
 /// Inject @p FlutterTextureRegistry and @p FlutterBinaryMessenger for unit testing.
-- (instancetype)initWithRegistry:(NSObject<FlutterTextureRegistry> *)registry
-                       messenger:(NSObject<FlutterBinaryMessenger> *)messenger;
+- (nonnull instancetype)initWithRegistry:(NSObject<FlutterTextureRegistry> *_Nullable)registry
+                               messenger:(NSObject<FlutterBinaryMessenger> *_Nullable)messenger;
 
 /// Inject @p FlutterTextureRegistry, @p FlutterBinaryMessenger, and Pigeon callback handler for
 /// unit testing.
-- (instancetype)initWithRegistry:(NSObject<FlutterTextureRegistry> *)registry
-                       messenger:(NSObject<FlutterBinaryMessenger> *)messenger
-                       globalAPI:(FCPCameraGlobalEventApi *)globalAPI NS_DESIGNATED_INITIALIZER;
+- (nonnull instancetype)initWithRegistry:(NSObject<FlutterTextureRegistry> *_Nullable)registry
+                               messenger:(NSObject<FlutterBinaryMessenger> *_Nullable)messenger
+                               globalAPI:(FCPCameraGlobalEventApi *_Nullable)globalAPI
```

Comment:
Are these three properties really nullable? I would expect things not to work without them.

Reply:
Yeah, you're right, they shouldn't be. That was a shortcut on my part that I forgot to remove later. It was easier to pass nil in tests, but those will need some mocking.

Reply:
I removed the nullability, to fix warnings in tests I had to introduce new some mocks for Flutter-related protocols.